tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1339,192 +1339,226 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
-				Port:              "4000",
-				TLSCluster:        true,
-				User:              "root",
-				Password:          "",
-			}
+		for _, sc := range cases {
+			ginkgo.It(sc.name, func() {
+				tcName := "tls"
 
-			ginkgo.By("Deploying tidb drainer")
-			err = oa.DeployDrainer(drainerConfig, source)
-			framework.ExpectNoError(err, "failed to deploy drainer: %v", drainerConfig)
-			err = oa.CheckDrainer(drainerConfig, source)
-			framework.ExpectNoError(err, "failed to check drainer: %v", drainerConfig)
+				ginkgo.By("Installing tidb CA certificate")
+				err := InstallTiDBIssuer(ns, tcName)
+				framework.ExpectNoError(err, "failed to install CA certificate")
 
-			ginkgo.By("Inserting data into source db")
-			err = wait.PollImmediate(time.Second*5, time.Minute*5, insertIntoDataToSourceDB(fw, c, ns, tcName, passwd, true))
-			framework.ExpectNoError(err, "insert data into source db timeout")
+				ginkgo.By("Installing tidb server and client certificate")
+				err = InstallTiDBCertificates(ns, tcName)
+				framework.ExpectNoError(err, "failed to install tidb server and client certificate")
 
-			ginkgo.By("Checking tidb-binlog works as expected")
-			err = wait.PollImmediate(time.Second*5, time.Minute*5, dataInClusterIsCorrect(fw, c, ns, targetTcName, "", false))
-			framework.ExpectNoError(err, "check data correct timeout")
+				ginkgo.By("Installing tidbInitializer client certificate")
+				err = installTiDBInitializerCertificates(ns, tcName)
+				framework.ExpectNoError(err, "failed to install tidbInitializer client certificate")
 
-			ginkgo.By("Connecting to tidb server to verify the connection is TLS enabled")
-			err = wait.PollImmediate(time.Second*5, time.Minute*5, tidbIsTLSEnabled(fw, c, ns, tcName, passwd))
-			framework.ExpectNoError(err, "connect to TLS tidb timeout")
+				ginkgo.By("Installing dashboard client certificate")
+				err = installPDDashboardCertificates(ns, tcName)
+				framework.ExpectNoError(err, "failed to install dashboard client certificate")
 
-			ginkgo.By("Scaling out tidb cluster")
-			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
-				tc.Spec.PD.Replicas = 5
-				tc.Spec.TiKV.Replicas = 5
-				tc.Spec.TiDB.Replicas = 3
-				tc.Spec.Pump.Replicas++
-				tc.Spec.TiFlash.Replicas = 3
-				tc.Spec.TiCDC.Replicas = 3
-				return nil
-			})
-			framework.ExpectNoError(err, "failed to update TidbCluster: %q", tc.Name)
-			err = oa.WaitForTidbClusterReady(tc, 30*time.Minute, 5*time.Second)
-			framework.ExpectNoError(err, "wait for TidbCluster ready timeout: %q", tc.Name)
+				ginkgo.By("Installing tidb components certificates")
+				err = InstallTiDBComponentsCertificates(ns, tcName)
+				framework.ExpectNoError(err, "failed to install tidb components certificates")
 
-			ginkgo.By("Scaling in tidb cluster")
-			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				ginkgo.By("Removing ca.crt in tidb-client-secret")
```

Comment:
Should also remove the ca from the secrets of initializer and dashboards when skipCA is set?